src/ccache.cpp

```diff
@@ -1036,8 +1161,47 @@ get_result_key_from_cpp(Context& ctx, Args& args, Hash& hash)
     stderr_path = tmp_stderr.path;
     ctx.register_pending_tmp_file(stderr_path);
 
+    TemporaryFile tmp_deps(FMT("{}/tmp.cpp_deps", ctx.config.temporary_dir()));
+    ctx.register_pending_tmp_file(tmp_deps.path);
+
     size_t args_added = 2;
     args.push_back("-E");
+
+    // Use -fminimize-whitespace if available. Outside unify mode, this will
+    // still normalize tokens on this same line.
+    if (ctx.config.compiler_type() == CompilerType::clang_minimize_whitespace
+        && language_supports_unify(ctx.args_info.actual_language)) {
+      args.push_back("-fminimize-whitespace");
+      args_added += 1;
+    }
```

Comment:
Column information is already inaccurate even without `-fminimize-whitespace` as detailed in the docs. What changes with `-fminimize-whitespace`:
 * Clang tries to preserve indentation of the first token. That is, your example would indeed show a different column, but only because it is the first token on line 3.
 * Clang emits a single space between two tokens on the same line if there was a space before the second token. With `-fminimize-whitespace`, a space is only added if required to not merge two tokens into one.